python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -2415,20 +2430,24 @@ def do_evaluate(
                 stream=stream,
             )
             if coalesce and how == "Full":
+                left_key_names = left_on.column_names
+                right_key_names = right_on.column_names
+                left_key_cols = left.select(left_key_names).column_map
+                right_key_cols = right.select(right_key_names).column_map
+
                 left = left.with_columns(
                     (
                         Column(
                             plc.replace.replace_nulls(
-                                left_col.obj, right_col.obj, stream=stream
+                                left_key_cols[name].obj, _rc.obj, stream=stream
                             ),
-                            name=left_col.name,
-                            dtype=left_col.dtype,
-                        )
-                        for left_col, right_col in zip(
-                            left.select_columns(left_on.column_names_set),
-                            right.select_columns(right_on.column_names_set),
-                            strict=True,
+                            name=name,
+                            dtype=left_key_cols[name].dtype,
                         )
+                        for name in left_key_names
```

Comment:
Is there a reason you can't iterate over `left_key_cols.items()` instead so that you don't have to index into `left_key_col` inside the `replace_null` call above?

Reply:
I reverted these changes and xfailed the polars test. I want to fix this in a separate PR.